code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -100,17 +116,23 @@
 			vampire.frenzy--
 
 		if (blood_total != vampire.blood_total)
-			var/update_msg = "<span class='notice'>You have accumulated [vampire.blood_total] [vampire.blood_total > 1 ? "units" : "unit"] of blood.</span>"
+			var/update_msg = "<span class='notice'>You have accumulated [vampire.blood_total] [vampire.blood_total > 1 ? "units" : "unit"] of blood</span>"
 			if (blood_usable != vampire.blood_usable)
-				update_msg += "<span class='notice'> And have [vampire.blood_usable] left to use.</span>"
+				update_msg += "<span class='notice'> and have [vampire.blood_usable] left to use.</span>"
+			else
+				update_msg += "<span class='notice'>.</span>"
 
 			to_chat(src, update_msg)
 		check_vampire_upgrade()
 		T.vessel.remove_reagent("blood", 25)
 
 	vampire.status &= ~VAMP_DRAINING
-	visible_message("<span class='danger'>[src.name] stops biting [T.name]'s neck!</span>", "<span class='notice'>You extract your fangs from [T.name]'s neck and stop draining them of blood. They will remember nothing of this occurance. Provided they survived.</span>")
-	if (T.stat != 2)
+
+	var/endsuckmsg = "You extract your fangs from [T.name]'s neck and stop draining them of blood."
+	if(vampire.stealth)
+		endsuckmsg += "They will remember nothing of this occurance, provided they survived."
+	visible_message("<span class='danger'>[src.name] stops biting [T.name]'s neck!</span>", "<span class='notice'>[endsuckmsg]</span>")
+	if (T.stat != 2 && vampire.stealth)
```

Comment:
You need to specify that they remember what happened imo. Otherwise they'll just assume they cant say anything since it might be a bug.